ThunkLibs/libX11/libX11_Guest.cpp

```diff
@@ -35,119 +38,198 @@ extern "C" {
 
 #include <vector>
 
+namespace {
+  constexpr static std::array<std::string_view, 13> CallbackKeys = {{
+    XNGeometryCallback,
+    XNDestroyCallback,
+    XNPreeditStartCallback,
+    XNPreeditDoneCallback,
+    XNPreeditDrawCallback,
+    XNPreeditCaretCallback,
+    XNPreeditStateNotifyCallback,
+    XNStatusStartCallback,
+    XNStatusDoneCallback,
+    XNStatusDrawCallback,
+    XNR6PreeditCallback,
+    XNStringConversionCallback,
+  }};
+
+  template<typename CallbackType>
+  static void ConvertArguments(std::vector<void*> &IncomingArguments, std::list<CallbackType> *Callbacks) {
+    // Walk the arguments and convert any callbacks.
```

Comment:
Should have an assert that `IncomingArguments.size()%2==0`

Reply:
Assert added.